addons/hr_homeworking_calendar/wizard/homework_location_wizard.xml

```diff
@@ -5,22 +5,42 @@
         <field name="model">homework.location.wizard</field>
         <field name="arch" type="xml">
             <form string="Location">
-                <group>
-                    <label for="date"/>
-                    <div class="o_row w-100">
-                        <field name="date"/>
-                    </div>
-                    <label for="weekly"/>
-                    <div class="o_row w-100">
-                        <field name="weekly" class="oe_inline"/>
+                <group col="12" class="mb-2">
+                    <group colspan="4">
+                        <i class="fa fa-clock-o ms-1" title="Date"/>
+                        <field name="date" class="min-w-2" nolabel="1"/>
+                    </group>
+                    <group colspan="8">
+                        <field name="weekly" class="oe_inline" nolabel="1" default_focus="1"/>
                         <span class="w-100">
                             Repeat every <field name="day_week_string" class="oe_inline"/>
                         </span>
+                    </group>
+                </group>
+                <group>
+                    <div class="o_row gap-2">
+                        <label for="work_location_id"/>
```

Comment:
Not bad :) usually, the standard structure to follow in a form view with some customized layout is the following:

```xml
<label for="your_field" class="oe_inline"/>
<div class="o_row">
    <field name="your_field" .../>
    (more stuff...)
</div>
```